packages/ui/src/components/vuestic-components/va-datepicker/components/VaDayPicker/VaDayPicker.vue

```diff
@@ -0,0 +1,268 @@
+<template>
+  <div class="va-date-picker-calendar">
+    <div class="va-date-picker-calendar__picker">
+      <div class="va-date-picker-calendar__weekdays weekdays" v-if="!hideWeekDays">
+        <div
+          v-for="weekday in weekdayNamesComputed" :key="weekday"
+          class="weekdays__weekday-cell"
+        >
+          <slot name="weekday">
+            {{ weekday }}
+          </slot>
+        </div>
+      </div>
+
+      <div class="va-date-picker-calendar__calendar calendar">
+        <div
+          class="calendar__day-wrapper"
+          v-for="date in calendarDates"
+          :key="date"
+          @mouseenter="hoveredDate = date"
+          @mouseleave="hoveredDate = null"
+        >
+          <div
+            class="calendar__day"
+            :class="{
+              'current-month': currentMonth === date.getMonth(),
+              'today': highlightTodayDate && isToday(date),
+              'current': isDateCurrentValue(date),
+              'in-range': isDateInRange(date),
+              'not-allowed': isDateNotAllowed(date),
+              'hightlighted-weekend': highlightWeekends && isDateWeekend(date)
+            }"
+            @click="onDateClick(date)"
+          >
+            <slot name="day" v-bind="{ date }">
+              {{ date.getDate() }}
+            </slot>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, PropType, toRefs, ref } from 'vue'
+import { useVaDatePickerCalendar } from './VaDatePickerCalendarHook'
+import { useHovered } from '../../hooks/HoveredOptionHook'
+import { isPeriod, isSingleDate, isDates } from '../../helpers/model-value-helper'
+import { isDatesArrayInclude, isDatesEqual } from '../../utils/date-utils'
+import { VaDatePickerModelValue } from '../../types/types'
+
+const isToday = (date: Date): boolean => date.toDateString() === new Date().toDateString()
+
+export default defineComponent({
+  name: 'VaDayPicker',
+
+  props: {
+    monthNames: { type: Array as PropType<string[]>, required: true, default: [] },
```

Comment:
No config support, which might be fine for  first iteration. But we want to have it eventually.

Reply:
VaDateHeader, VaDayPicker, and VaMonthPicker are private components, that VaDatePicker uses. If the user wants to change `month-names` he can use such components config:

```ts
...
VaDatePicker: { monthNames: [...] }
...
```

Almost all props from VaDateHeader, VaDayPicker, and VaMonthPicker are required and should be passed from the VaDatePicker component, there is why users need to configure only VaDatePicker.